src/mpid/ch4/shm/posix/posix_coll_select.h

```diff
@@ -101,6 +101,27 @@ MPIDI_POSIX_coll_algo_container_t *MPIDI_POSIX_Reduce_select(const void *sendbuf
     MPI_Aint type_size = 0;
     int pof2 = 0;
 
+    if (MPIDI_POSIX_Reduce_algo_choice == MPIDI_POSIX_Reduce_intra_release_gather_id &&
+        MPIR_Op_is_commutative(op)) {
+        /* release_gather based algorithm can be used only if izem submodule is built (and enabled)
+         * and MPICH is not multi-threaded. Also when the op is commutative */
+#ifdef ENABLE_IZEM_ATOMIC
```

Comment:
I think we're getting to the point where it makes sense to build and include izem by default. I'm not crazy about having to guard this code. @halimamer does build izem with all features automatically enable them in MPICH? Or are there additional configuration options?

Reply:
There are still problems building all of the izem features on MacOS. Before we can enable it, we need to resolve those issues.

Reply:
Got it. I made https://github.com/pmodels/izem/issues/20 as a reminder.

Reply:
Also, @halimamer had mentioned that izem performance is horrible when we oversubscribe the cores with threads.  It might not be a common-case in HPC, but jenkins will go bonkers.

Reply:
My proposal is to build with `izem=atomic` by default, so that the intra-node collectives can be used without any special configure option. This code would still need to be protected with izem ifdefs in case `izem=atomic` was disabled.

Reply:
Yes, `--enable-izem=atomic` by default sounds reasonable. The only question at that point is why have 2 atomic libraries? Obviously outside the scope of this PR, but I think we need to come up with a plan for izem vs. openpa atomics long-term.

Reply:
Did you see my comment, @raffenet ?

Reply:
Are you saying izem atomic performance is horrible in oversubscribed cases? Aren't they just wrappers around C11 atomics?

Reply:
I believe that's what @halimamer told me.  He'll need to confirm, but IIRC he said it's so bad that all the tests simply timeout.

Reply:
I ran the tests by enabling `izem=atomic` as part of this PR. The tests passed as they should. As far as I know, `izem=queue or sync` results in bad performance.

Reply:
I guess the oversubscribed case is the disputed one, though? It still seems odd because there is no code outside of this PR that uses izem atomics in MPICH.

Reply:
Oh nevermind, I think I misunderstood. Anyway, we can discuss this outside the context of this PR.

Reply:
The `izem` atomics themselves shouldn't have any performance negative side effects. They are just wrappers around GCC __atomic or C11 atomic operations. The oversubscription issue only happens when you busy wait without going to the OS-kernel, in which case calling `sched_yield()` or using POSIX synchronization primitives is more appropriate